tests/core/test_utilities.py

```diff
@@ -1850,3 +1851,13 @@ def test_parse_interaction_event_raises_wrong_type():
         match='.*either a str or.*vtk.vtkCommand.EventIds.*int.* was given',
     ):
         _parse_interaction_event(1)
+
+
+def test_classproperty():
+    class Foo:
+        @_classproperty
+        def prop(cls):
+            return 42
+
+    assert Foo.prop == 42
+    assert Foo().prop == 42
```

Comment:
Should we also test that `Foo.prop()` and the instance variation both fail?

Reply:
This implementation doesn't off any guards against setting the property. You can do `Foo.prop = 99` without error... not a perfect workaround

Reply:
Hmmmmm.... This is a decently big downside.  Without the callable nature and restricting setting, someone may be tempted to do `Theme.defaults = {`my_default`}` to register a default theme, in regards to #5119.

There is a suggestion that a metaclass is required https://stackoverflow.com/a/76047344

I'm not that familiar with metaclasses, but I'm pretty sure that this would be a one at a time type of usage rather than an easy to use decorator like here.  Maybe best just to keep this PR as is, and I will add in a sentence about registering themes into the `default` docstring.